sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -95,7 +120,7 @@ def __init__(self, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
-    def fit(self, X, y=None):
+    def fit(self, X, y=None, validate_types=True):
```

Comment:
I was suggesting more like we could maybe just remove this param altogether as it is the desired behaviour (right?)